src/js/other-websites/inpage_menu.js

```diff
@@ -356,100 +489,115 @@ const buildContent = {
+      // If so, we'll end up building two discret mask lists:  
+      // 1. Just masks for that specific website
+      // 2. All masks (including ones from the previous list)
+      const userHasMasksForCurrentWebsite = (checkIfAnyMasksWereGeneratedOnCurrentWebsite(masks, currentPageHostName)) || haveMasksBeenUsedOnCurrentSite(masks, currentPageHostName);
 
-      } else if (masks.length > 5) {
+      // If there are no masks assosiated with the current site, remove that list entirely.
+      if (!userHasMasksForCurrentWebsite) {
+        document.querySelector(".fx-relay-menu-masks-list-this-website").remove();
+      }
 
-        // If there's at least 6 masks, show the search bar
-        await populateMaskList(searchResults, masks, {
-          replaceMaskAddressWithLabel: true,
-        });
+      const maskLists = document.querySelectorAll(".fx-relay-menu-masks-list");
 
-        searchResultsList.style.height = `${searchResultsList.offsetHeight}px`;
+      // Now we can set the toggle buttons between the two lists.
+      if (userHasMasksForCurrentWebsite) {
+        fxRelayMenuBody.classList.add(".fx-relay-mask-list-toggle-height");
 
-        const filterSearchForm = document.querySelector(
-          ".fx-relay-menu-masks-search-form"
+        const filterMenu = document.querySelector(
+          ".fx-relay-menu-filter-active-site"
         );
+        filterMenu.classList.add("is-visible");
 
-        const filterSearchInput = filterSearchForm.querySelector(
-          ".fx-relay-menu-masks-search-input"
-        );
+        const filterMenuButtons = filterMenu.querySelectorAll("button");
 
-        filterSearchForm.addEventListener("submit", (event) => {
-          event.preventDefault();
-          filterSearchInput.blur();
-        });
+        filterMenuButtons.forEach((button) => {
+          const stringId = button.dataset.stringId;
+          button.textContent = browser.i18n.getMessage(stringId);
 
-        filterSearchInput.addEventListener("input", (event) => {
-          applySearchFilter(event.target.value);
+          // TODO: Move this function elsewhere
+
+          button.addEventListener("click", async (event) => {
+            await buildContent.components.setMaskListButton(event.target, maskLists, filterMenuButtons);
+          });
         });
+      }
 
-      } else {
-        // User has between 1-5 masks. Display all of them, but
-        // do not show the search input/filter.
+      maskLists?.forEach(async (maskList) => {
 
-        await populateMaskList(searchResults, masks, {
-          replaceMaskAddressWithLabel: true,
-        });
+        // Check if the list we're currently building is "From this website". This logic will be used throughout the function
+        const buildFilteredMaskList = maskList.classList.contains(
+          "fx-relay-menu-masks-list-this-website"
+        );
+
+        // TODO: Check additional field(s) besides "generated_for"
```

Comment:
`TODONE`!

Reply:
Removed in d5e6221fe717faa6809d549a9e0b85259f67a063